classes/local/store/s3/client.php

```diff
@@ -235,49 +288,54 @@ public function test_permissions($testdelete) {
         $permissions->success = true;
         $permissions->messages = array();
 
-        try {
-            $result = $this->client->putObject(array(
-                            'Bucket' => $this->bucket,
-                            'Key' => $this->bucketkeyprefix . 'permissions_check_file',
-                            'Body' => 'test content'));
-        } catch (\Aws\S3\Exception\S3Exception $e) {
-            $details = $this->get_exception_details($e);
-            $permissions->messages[get_string('settings:writefailure', 'tool_objectfs') . $details] = 'notifyproblem';
+        if (!$this->is_functional()) {
             $permissions->success = false;
-        }
-
-        try {
-            $result = $this->client->getObject(array(
-                            'Bucket' => $this->bucket,
-                            'Key' => $this->bucketkeyprefix . 'permissions_check_file'));
-        } catch (\Aws\S3\Exception\S3Exception $e) {
-            $errorcode = $e->getAwsErrorCode();
-            // Write could have failed.
-            if ($errorcode !== 'NoSuchKey') {
+            $permissions->messages = array();
+        } else {
+            try {
+                $result = $this->client->putObject(array(
+                    'Bucket' => $this->bucket,
+                    'Key' => $this->bucketkeyprefix . 'permissions_check_file',
+                    'Body' => 'test content'));
+            } catch (\Aws\S3\Exception\S3Exception $e) {
                 $details = $this->get_exception_details($e);
-                $permissions->messages[get_string('settings:readfailure', 'tool_objectfs') . $details] = 'notifyproblem';
+                $permissions->messages[get_string('settings:writefailure', 'tool_objectfs') . $details] = 'notifyproblem';
                 $permissions->success = false;
             }
-        }
 
-        if ($testdelete) {
             try {
-                $result = $this->client->deleteObject(array('Bucket' => $this->bucket, 'Key' => $this->bucketkeyprefix . 'permissions_check_file'));
-                $permissions->messages[get_string('settings:deletesuccess', 'tool_objectfs')] = 'warning';
-                $permissions->success = false;
+                $result = $this->client->getObject(array(
+                    'Bucket' => $this->bucket,
+                    'Key' => $this->bucketkeyprefix . 'permissions_check_file'));
             } catch (\Aws\S3\Exception\S3Exception $e) {
                 $errorcode = $e->getAwsErrorCode();
-                // Something else went wrong.
-                if ($errorcode !== 'AccessDenied') {
+                // Write could have failed.
```

Comment:
This comment seems misplaced (maybe it belongs in the previous block for `writefailure`

This is indicating a read failure I believe.

Reply:
removing completely.

Reply:
removed the comment